library/src/main/java/org/xmtp/android/library/Client.kt

```diff
@@ -181,18 +194,18 @@ class Client() {
             }
         }
 
-        return apiClient.publish(envelopes = envelopes)
+        return runBlocking { apiClient.publish(envelopes = envelopes) }
     }
 
-    suspend fun ensureUserContactPublished() {
+    fun ensureUserContactPublished() {
         address?.let {
             val contact = getUserContact(peerAddress = it)
             if (contact != null && keys?.getPublicKeyBundle() == contact.v2.keyBundle) {
                 return
             }
         }
 
-        publishUserContact(legacy = true)
+        runBlocking { publishUserContact(legacy = true) }
```

Comment:
Does this need to be in a `runBlocking` if it's not a `suspend` any more? Just asking out of curiosity